src/ngraph/runtime/cpu/pass/cpu_mkldnn_primitive_build.cpp

```diff
@@ -0,0 +1,814 @@
+    {TI(ConvolutionBias), &MKLDNNPrimitiveBuildPass::build_primitive<ConvolutionBias>},
+    {TI(QuantizedConvolution), &MKLDNNPrimitiveBuildPass::build_primitive<QuantizedConvolution>},
+    {TI(ConvolutionBiasBackpropFiltersBias),
+     &MKLDNNPrimitiveBuildPass::build_primitive<ConvolutionBiasBackpropFiltersBias>},
+    {TI(LRN), &MKLDNNPrimitiveBuildPass::build_primitive<LRN>},
+    {TI(Relu), &MKLDNNPrimitiveBuildPass::build_primitive<Relu>},
+    {TI(ReluBackprop), &MKLDNNPrimitiveBuildPass::build_primitive<ReluBackprop>},
+    {TI(LeakyRelu), &MKLDNNPrimitiveBuildPass::build_primitive<LeakyRelu>},
+    {TI(Sigmoid), &MKLDNNPrimitiveBuildPass::build_primitive<Sigmoid>},
+    {TI(SigmoidBackprop), &MKLDNNPrimitiveBuildPass::build_primitive<SigmoidBackprop>},
+    {TI(Lstm), &MKLDNNPrimitiveBuildPass::build_primitive<Lstm>},
+    {TI(Rnn), &MKLDNNPrimitiveBuildPass::build_primitive<Rnn>},
+    {TI(QuantizedMaxPool), &MKLDNNPrimitiveBuildPass::build_primitive<QuantizedMaxPool>},
+    {TI(QuantizedAvgPool), &MKLDNNPrimitiveBuildPass::build_primitive<QuantizedAvgPool>},
+    {TI(Softmax), &MKLDNNPrimitiveBuildPass::build_primitive<Softmax>},
+    {TI(Slice), &MKLDNNPrimitiveBuildPass::build_primitive<Slice>},
+    {TI(ReplaceSlice), &MKLDNNPrimitiveBuildPass::build_primitive<ReplaceSlice>},
+    {TI(UpdateSlice), &MKLDNNPrimitiveBuildPass::build_primitive<UpdateSlice>},
+    {TI(ConvolutionAdd), &MKLDNNPrimitiveBuildPass::build_primitive<ConvolutionAdd>},
+    {TI(QuantizedConvolutionRelu),
+     &MKLDNNPrimitiveBuildPass::build_primitive<QuantizedConvolutionRelu>},
+    {TI(QuantizedConvolutionBias),
+     &MKLDNNPrimitiveBuildPass::build_primitive<QuantizedConvolutionBias>},
+    {TI(QuantizedConvolutionBiasAdd),
+     &MKLDNNPrimitiveBuildPass::build_primitive<QuantizedConvolutionBiasAdd>},
+    {TI(QuantizedConvolutionBiasSignedAdd),
+     &MKLDNNPrimitiveBuildPass::build_primitive<QuantizedConvolutionBiasSignedAdd>},
+    {TI(GroupConvolutionBias), &MKLDNNPrimitiveBuildPass::build_primitive<GroupConvolutionBias>},
+    {TI(Quantize), &MKLDNNPrimitiveBuildPass::build_primitive<Quantize>},
+    {TI(Dequantize), &MKLDNNPrimitiveBuildPass::build_primitive<Dequantize>},
+    {TI(QuantizedConcat), &MKLDNNPrimitiveBuildPass::build_primitive<QuantizedConcat>},
+    {TI(GetOutputElement), &MKLDNNPrimitiveBuildPass::build_primitive<GetOutputElement>},
+};
+
+bool MKLDNNPrimitiveBuildPass::run_on_call_graph(const std::list<std::shared_ptr<Node>>& nodes)
+{
+    for (const auto& shp_node : nodes)
+    {
+        Node* node = shp_node.get();
+
+        if (mkldnn_utils::use_mkldnn_kernel(node))
+        {
+            auto handler = prim_build_dispatcher.find(TI(*node));
+            NGRAPH_ASSERT(handler != prim_build_dispatcher.end())
+                << "Unsupported node '" << node->description() << "' in MKLDNNPrimitiveBuildPass";
+
+            handler->second(m_mkldnn_emitter, node);
```

Comment:
Minor nit but m_mkldnn_emitter.m_node_primitive_index_map[node] = handler->second(m_mkldnn_emitter, node) seems cleaner than doing it in all the builder methods.